fuzz/fuzz_targets/selection_parsing.rs

```diff
@@ -0,0 +1,112 @@
+#![no_main]
+
+use std::sync::LazyLock;
+
+use apollo_federation::sources::connect::JSONSelection;
+use bnf::Grammar;
+use libfuzzer_sys::arbitrary;
+use libfuzzer_sys::arbitrary::Arbitrary;
+use libfuzzer_sys::fuzz_target;
+use libfuzzer_sys::Corpus;
+use rand::rngs::StdRng;
+
+fuzz_target!(|input: GeneratedSelection| -> Corpus {
+    // Generating a selection might choose a path which recurses too deeply, so
+    // we just mark those traversals as being rejected since they would require
+    // seeding and iterating the Rng.
+    let Some(selection) = input.0 else {
+        return Corpus::Reject;
+    };
+
+    let parsed = JSONSelection::parse(&selection).unwrap();
+    drop(parsed);
+
+    Corpus::Keep
+});
```

Comment:
this thought would have to be a different target entirely and is non-blocking: it would interesting to see how this would behave in addition to a generated supergraph and operation. this only fuzzes a specific part of a whole - but how will it behave all together?